### ISSUE TITLE
установить флаг strict: true в tsconfig.json

### ISSUE THREAD

**Issue by sk1e** [fullstack-development/react-redux-starter-kit#164]
почему strictFunctionTypes сейчас выставлен в false?

**Comment by in19farkt** [comment]
там вообще нужно убрать эти флаги
```
noImplicitAny
strictNullChecks
strictFunctionTypes
strictBindCallApply
strictPropertyInitialization
noImplicitThis
alwaysStrict
```
и поставить один, который всё это объединяет`"strict": true,`

**Comment by in19farkt** [comment]
И еще эти нужно включить (если они не включены)
```
    /* Additional Checks */
    "noUnusedLocals": true,                /* Report errors on unused locals. */
    "noUnusedParameters": true,            /* Report errors on unused parameters. */
    "noImplicitReturns": true,             /* Report error when not all code paths in function return a value. */
    "noFallthroughCasesInSwitch": true,    /* Report errors for fallthrough cases in switch statement. */
```

**Comment by Znack** [comment]
Поправил название для `"strict": true`, @sk1e норм же?

**Comment by sk1e** [comment]
> Поправил название для `"strict": true`, @sk1e норм же?

да, я просто выяснить хотел почему именно strictFunctionTypes в false выставлен был. Выглядит преднамеренно

**Comment by in19farkt** [comment]
> да, я просто выяснить хотел почему именно strictFunctionTypes в false выставлен был. Выглядит преднамеренно

Когда только этот флаг появился, были какие-то проблемы, не сходились типы там где они должны были сходиться, и мы решили вырубить его. Возможно в следующих версиях это пофиксили, т.к. сейчас я не испытываю никаких трудностей с этим флагом.

**Comment by safr** [comment]
Лучше добавлять флаги отдельно, так мы внесем ясность что происходит, и с strict:true есть проблемы, иногда мы знаем больше о коде , чем тс, но так как все в строгом режиме он нам будет ошибки сыпать
